### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by dngroth**

### CODE REVIEW

File: PgpWordList.Test/PgpWordAggregationParserTest.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using Xunit;
+
+namespace Messerli.PgpWordList.Test
+{
+    public class PgpWordAggregationParserTest
+    {
+        [Theory]
+        [ClassData(typeof(PgpWordListTestData))]
+        public void ParsePgpWordAccumulation(byte[] expectedPgpBytes, string separator, string pgpString)
```
Comment (janhohenheim) [inline PgpWordList.Test/PgpWordAggregationParserTest.cs]: Make this a describtion of what _should_ happen

File: PgpWordList/PgpWordAggregationBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
+    public class PgpWordAggregationBuilder
+    {
+        public const string DefaultSeparator = "-";
+
+        private readonly List<byte> _byteList = new List<byte>();
+
+        private string _separator = DefaultSeparator;
+
+        public PgpWordAggregationBuilder SetSeparator(string separator)
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationBuilder.cs]: We usually omit the `Set` in builders.

File: PgpWordList/PgpWordAggregationBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        public PgpWordAggregation Build()
+            => new PgpWordAggregation(
+                string.Join(
+                    _separator,
+                    _byteList.Aggregate(new List<string>(), ByteToWord)));
+
+        private static List<string> ByteToWord(List<string> list, byte @byte)
+        {
+            list.Add(
+                list.Count % 2 == 0
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationBuilder.cs]: Extract this into an `IsEven` function

File: PgpWordList/PgpWordAggregationBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Messerli.PgpWordList
+{
+    public class PgpWordAggregationBuilder
+    {
+        public const string DefaultSeparator = "-";
+
+        private readonly List<byte> _byteList = new List<byte>();
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationBuilder.cs]: Could you use an `ImmutableList`?

File: PgpWordList/PgpWordAggregationParser.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        public byte[] Parse(PgpWordAggregation pgpWordAggregation)
+            => pgpWordAggregation
+                .Value
+                .Split(_separator)
+                .Aggregate(new List<byte>(), WordToByte)
+                .ToArray();
+
+        private static List<byte> WordToByte(List<byte> list, string word)
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationParser.cs]: This method does more than just convert a word to a byte, as it also appends it to a list.

File: PgpWordList/PgpWordAggregationParser.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            var @byte = list.Count % 2 == 0
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationParser.cs]: Extract into an `IsEven` function

File: PgpWordList/PgpWordList.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            => Data[@byte].Even;
+
+        public static string MapOdd(byte @byte)
+            => Data[@byte].Odd;
+
+        public static byte? FindEven(string word)
+            => Data.FirstOrDefault(element => element.Value.Even.Equals(word)).Key;
+
+        public static byte? FindOdd(string word)
+            => Data.FirstOrDefault(element => element.Value.Odd.Equals(word)).Key;
```
Comment (janhohenheim) [inline PgpWordList/PgpWordList.cs]: Avoid code duplication, use a higher order function.

File: PgpWordList/PgpWordListData.cs [hunk 8]
```diff
@@ -0,0 +1,271 @@
+using System.Collections.Generic;
+
+namespace Messerli.PgpWordList
+{
+    /// <summary>
+    /// Contains the word list from https://en.wikipedia.org/wiki/PGP_word_list.
```
Comment (janhohenheim) [inline PgpWordList/PgpWordListData.cs]: Link the URL with `<a href="...">Foo</a>`

File: PgpWordList/PgpWordAggregationExtension.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Linq;
+
+namespace Messerli.PgpWordList
+{
+    public static class PgpWordAggregationExtension
```
Comment (janhohenheim) [inline PgpWordList/PgpWordAggregationExtension.cs]: I'm not a fan of making the business code of this library into extension functions for primitive data types. The extension function on ``PgpWordAggregation` in particular seems a roundabout equivalent of putting functionality into the POD directly.

The code presented here is pretty similar to [System.BitConverter](https://docs.microsoft.com/en-us/dotnet/api/system.bitconverter?view=netframework-4.8). How about extracting this functionality into a class like `PgpWordBitConverter`?

File: PgpWordList/PgpWordList.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
+using System.Linq;
+
+namespace Messerli.PgpWordList
+{
+    public static partial class PgpWordList
```
Comment (janhohenheim) [inline PgpWordList/PgpWordList.cs:6]: clever!

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: PgpWordList.Test/PgpWordAggregationParserTest.cs (hunk 1 above)
Comment (dngroth) [reply PgpWordList.Test/PgpWordAggregationParserTest.cs]: Addressed in https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/ee7e0ee16c845190597975a16154bd59ab465523

File: PgpWordList/PgpWordAggregationBuilder.cs (hunk 3 above)
Comment (dngroth) [reply PgpWordList/PgpWordAggregationBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/048fb2d105c4bf7d629d2da2b6273312daae5708

File: PgpWordList/PgpWordAggregationParser.cs (hunk 6 above)
Comment (dngroth) [reply PgpWordList/PgpWordAggregationParser.cs]: Addressed https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/048fb2d105c4bf7d629d2da2b6273312daae5708

File: PgpWordList/PgpWordAggregationParser.cs (hunk 5 above)
Comment (dngroth) [reply PgpWordList/PgpWordAggregationParser.cs]: Addressed https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/ec053a2ee9523187cd2a046dbb6e7e605233b95e

File: PgpWordList/PgpWordAggregationBuilder.cs (hunk 4 above)
Comment (dngroth) [reply PgpWordList/PgpWordAggregationBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/182ab8e29afaf17e028f101f49b1143780742f7b, https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/84783867218fd7fc5345719729655caeeba65565

File: PgpWordList/PgpWordList.cs (hunk 7 above)
Comment (dngroth) [reply PgpWordList/PgpWordList.cs]: addressed https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/003b7207fb4a6a2cdac937a532f93e13355ebfe8

File: PgpWordList/PgpWordListData.cs (hunk 8 above)
Comment (dngroth) [reply PgpWordList/PgpWordListData.cs]: addressed https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/3d124497e428d3e66501bbe1ae58f200471abf31

File: PgpWordList/PgpWordAggregationExtension.cs (hunk 9 above)
Comment (dngroth) [reply PgpWordList/PgpWordAggregationExtension.cs]: addressed https://github.com/messerli-informatik-ag/pgp-word-list/pull/1/commits/7ac55c2b572751086bff256d7a56e80963a28a04

**Review by janhohenheim — APPROVED** [review]

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Add our default project settings
- coding style
- nullability etc.

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**